crates/driver/src/domain/competition/auction.rs

```diff
@@ -90,17 +86,28 @@ impl From<Deadline> for chrono::DateTime<chrono::Utc> {
 }
 
 #[derive(Debug, Clone, Copy)]
-pub struct Id(pub u64);
+pub struct Id(i64);
 
 impl Id {
+    // The largest valid auction ID.
+    pub const MAX: i64 = i64::MAX;
+    // The smallest valid auction ID.
+    pub const MIN: i64 = 0;
+
     pub fn to_be_bytes(self) -> [u8; 8] {
         self.0.to_be_bytes()
     }
 }
 
-impl From<u64> for Id {
-    fn from(inner: u64) -> Self {
-        Self(inner)
+impl TryFrom<i64> for Id {
+    type Error = InvalidId;
+
+    fn try_from(value: i64) -> Result<Self, Self::Error> {
+        if value >= 0 {
```

Comment:
Since this PR introduces these constants they should probably be used.
On the other hand why was the `Id` changed from `i64` to `u64` in the first place?
Is this because ids are an `i64` on the postgres level? I'm not sure we have to concern ourselves with that since they won't be negative anyway.
```suggestion
        if (Self::MIN..=Self::MAX).contains(&value) {
```

Reply:
> possibly JSON doesn't support them either

[Yep](https://stackoverflow.com/a/34989371). Man, JS is weird. 😅

Reply:
Autopilot sends an i64 because on the Postgres level it is an i64. It is intended to always be positive and it won't reach i64::MAX so which of the two Rust types you use doesn't matter so much.

The Json standard doesn't say how numbers work. They are just a sequence of digits. The browser Javascript engines always treat them as f64 as Martin linked. If you send Json betwen Rust code using serde derives on a struct with an u64 then that works.